hudi-common/src/main/java/org/apache/hudi/common/util/collection/ExternalSpillableMap.java

```diff
@@ -61,15 +63,17 @@
   private final long maxInMemorySizeInBytes;
   // Map to store key-values in memory until it hits maxInMemorySizeInBytes
   private final Map<T, R> inMemoryMap;
-  // Map to store key-valuemetadata important to find the values spilled to disk
-  private transient volatile DiskBasedMap<T, R> diskBasedMap;
+  // Map to store key-values on disk or db after it spilled over the memory
+  private transient volatile SpillableDiskMap<T, R> diskBasedMap;
```

Comment:
does n't `DiskBasedMap` already mean its spilling to disk. May be a better hierarchy is to call the new interface
`DiskMap` (instead of SpillableDiskMap) and rename DiskBasedMap to `BitCaskDiskMap` (which is what the design is based on) https://github.com/basho/bitcask